### PR TITLE
Upgrade to cdk version 0.26.0

### PR DESCRIPTION
**Pull request by jacobwinch**
## What does this change?

This PR updates the version of the `@guardian/cdk` library and makes use of a [new scheduled lambda pattern](https://github.com/guardian/cdk/blob/main/src/patterns/scheduled-lambda.ts) for the first time. 

This doesn't actually simplify our infrastructure definition much _yet_, but it allows us to configure an appropriate alert with minimal boilerplate in the future (I'll do this in a subsequent PR). It also means that we start conforming to [the recommendation](https://github.com/guardian/cdk#patterns) in the `@guardian/cdk` README:

`Patterns should be your default entry point to this library.`

## How to test

I think [this check](https://github.com/guardian/tag-janitor/blob/main/.github/workflows/ci.yml) should suffice?

## How can we measure success?

Moving to pattern (rather than construct) usage should help us to keep up to date with best practices more easily in the future.

### CODE REVIEW

File: cdk/lib/__snapshots__/cdk-stack.test.ts.snap [hunk 1]
```diff
@@ -321,7 +321,6 @@ Object {
     },
     "tagjanitorlambdatagjanitorlambdarate7days0B5F68379": Object {
       "Properties": Object {
-        "Description": "Run tag-janitor every 7 days",
```
Comment (jacobwinch) [inline cdk/lib/__snapshots__/cdk-stack.test.ts.snap:324]: This has been dropped as [the new pattern doesn't take a description](https://github.com/guardian/cdk/blob/main/src/patterns/scheduled-lambda.ts#L8-L11). I think this is pretty clear from the `rate(7 days)` and the `Targets` block anyway.

File: cdk/lib/cdk-stack.ts [hunk 2]
```diff
@@ -52,16 +52,12 @@ export class CdkStack extends GuStack {
...
         subnets: GuVpc.subnets(this, parameters.subnets.valueAsList),
       },
-      rules: [
-        {
-          schedule: Schedule.rate(lambdaFrequency),
-          description: `Run tag-janitor every ${lambdaFrequency.toHumanString()}`,
-        },
-      ],
+      schedule: Schedule.rate(lambdaFrequency),
+      monitoringConfiguration: { noMonitoring: true },
```
Comment (jacobwinch) [inline cdk/lib/cdk-stack.ts:53]: This PR is a simple upgrade, so I won't add new resources here; I will add an alert via monitoringConfiguration in a subsequent PR.

File: cdk/package.json [hunk 3]
```diff
@@ -13,13 +13,13 @@
...
   "devDependencies": {
-    "@aws-cdk/assert": "~1.74.0",
+    "@aws-cdk/assert": "1.86.0",
     "@guardian/eslint-config-typescript": "^0.4.1",
     "@types/jest": "^26.0.20",
     "@types/node": "10.17.27",
     "@typescript-eslint/eslint-plugin": "^4.14.0",
     "@typescript-eslint/parser": "^4.14.0",
-    "aws-cdk": "~1.74.0",
+    "aws-cdk": "1.86.0",
```
Comment (jacobwinch) [inline cdk/package.json:22]: These version bumps are required to keep us in sync after https://github.com/guardian/cdk/pull/191.

File: cdk/lib/cdk-stack.ts [hunk 4]
```diff
@@ -52,16 +52,12 @@ export class CdkStack extends GuStack {
       description: "Lambda to notify about instances with missing tags",
       timeout: Duration.seconds(30),
       memorySize: 512,
-      vpc: GuVpc.fromId(this, "vpc", parameters.vpc.valueAsString),
+      vpc: GuVpc.fromId(this, "vpc", { vpcId: parameters.vpc.valueAsString }),
```
Comment (jacobwinch) [inline cdk/lib/cdk-stack.ts]: This was changed [here](https://github.com/guardian/cdk/pull/149/files#diff-167082b85b20aeff983069d8655e08f9fa824bc77db741afd8fed1afcbb536a2R17).

Comment (jamie-lynch) [reply cdk/lib/cdk-stack.ts]: There's also now the `GuVpc.fromIdParameter` and `GuVpc.subnetsFromParameter` methods which would save defining those parameters explicitly. It looks up the values from parameter store so they may need adding there first.

File: cdk/lib/cdk-stack.ts [hunk 5]
```diff
@@ -52,16 +52,12 @@ export class CdkStack extends GuStack {
       description: "Lambda to notify about instances with missing tags",
       timeout: Duration.seconds(30),
       memorySize: 512,
```
Comment (jamie-lynch) [inline cdk/lib/cdk-stack.ts]: Can these two props be omitted now as they match the default values as of https://github.com/guardian/cdk/pull/188?

Comment (jacobwinch) [reply cdk/lib/cdk-stack.ts]: Yep, good point, thanks for spotting this 👍

File: cdk/lib/cdk-stack.ts (hunk 4 above)
Comment (jacobwinch) [reply cdk/lib/cdk-stack.ts]: Cool, I'll set this up now.

**Review by jamie-lynch — APPROVED** [review]

**Pull request merged**